chain/db.go

```diff
@@ -807,7 +806,14 @@ func NewDBStore(db DB, n *consensus.Network, genesisBlock types.Block) (_ *DBSto
 		if err := dbs.Flush(); err != nil {
 			return nil, consensus.State{}, err
 		}
-	} else if dbGenesis.ID != genesisBlock.ID() {
+	} else if version[0] != 2 {
+		if err := MigrateDB(db, n, noopLogger{}); err != nil {
```

Comment:
Isn't it worth passing in a logger to `NewDBStore` to make us aware a migration is taking place? Definitely a little annoying but the way it is now this would hang for a really long time without any visible trace of what's going on. 
User might decide to cancel out and would have to start from scratch every time.

Reply:
+1 I'd consider Nate's suggested approach of a functional argument for a zap logger

Reply:
If we use a functional argument, then the API won't break, but you still need to update all your callsites to pass a logger. I don't see how that's much better than updating all your callsites to call `MigrateDB`.
I'm not really a fan of functional arguments in general; they remove the "cost" of adding new parameters to a function, without which things tend to complexify without bound.

Anyway, the important thing is to get this merged so that Zen can stop being broken, so I'll do whatever you guys feel is most expedient.

Reply:
I don't mind breaking the signature but I do vote for the migration being implicit and probably for the no-op logger to be exposed. It's quite annoying to have a dependency break and to find out you need to implement some interface to satisfy a function call. Might as well make it painless for people who don't care about logging.

Regarding complexity I'd argue that figuring out a function argument to enable logging has lower complexity than figuring out you need to call a separate method to do the same 😅